src/NonFungibleToken/NonFungibleToken/NonFungibleToken.cs

```diff
@@ -0,0 +1,491 @@
+        /// <returns>A value indicating if the operator has permissions to act on behalf of the owner.</returns>
+        private bool GetOwnerToOperator(Address owner, Address operatorAddress)
+        {
+            return this.PersistentState.GetBool($"OwnerToOperator:{owner}:{operatorAddress}");
+        }
+
+        /// <summary>
+        /// Sets the owner to operator permission.
+        /// </summary>
+        /// <param name="owner">The owner address of the NFT.</param>
+        /// <param name="operatorAddress">>Address to add to the set of authorized operators.</param>
+        /// <param name="value">The permission value.</param>
+        private void SetOwnerToOperator(Address owner, Address operatorAddress, bool value)
+        {
+            this.PersistentState.SetBool($"OwnerToOperator:{owner}:{operatorAddress}", value);
+        }
+
+        /// <summary>
+        /// Constructor. Initializes the supported interfaces.
+        /// </summary>
+        /// <param name="state">The smart contract state.</param>
+        public NonFungibleToken(ISmartContractState state) : base(state)
+        {
+            // todo: discuss callback handling and supported interface numbering with community.
+            this.SetSupportedInterfaces((uint)0x00000001, true); // (ERC165) - ISupportsInterface
+            this.SetSupportedInterfaces((uint)0x00000002, true); // (ERC721) - INonFungibleToken,
+            this.SetSupportedInterfaces((uint)0x00000003, true); // (ERC721) - INonFungibleTokenReceiver
+        }
+
+        /// <summary>
+        /// Function to check which interfaces are supported by this contract.
+        /// </summary>
+        /// <param name="interfaceID">Id of the interface.</param>
+        /// <returns>True if <see cref="interfaceID"/> is supported, false otherwise.</returns>
+        public bool SupportsInterface(uint interfaceID)
+        {
+            return GetSupportedInterfaces(interfaceID);
+        }
+
+        /// <summary>
+        /// Handle the receipt of a NFT. The smart contract calls this function on the
+        /// recipient after a transfer. This function MAY throw or return false to revert and reject the transfer.
+        /// Return true if the transfer is ok.
+        /// </summary>
+        /// <remarks>Up to the contract to implement. This contract just returns true.</remarks>
+        /// <param name="operatorAddress">The address which called safeTransferFrom function.</param>
+        /// <param name="fromAddress">The address which previously owned the token.</param>
+        /// <param name="tokenId">The NFT identifier which is being transferred.</param>
+        /// <param name="data">Additional data with no specified format.</param>
+        /// <returns>A bool indicating the resulting operation.</returns>
+        public bool OnNonFungibleTokenReceived(Address operatorAddress, Address fromAddress, ulong tokenId, byte[] data)
+        {            
+            return true;
```

Comment:
Missing some logic here?

Reply:
On eth there is a keccak hash based byte4 reply but I found that to be overkill so emptied out the method, if we need to do something here I'll be happy to add it.

Reply:
Is there a purpose to keeping it at all then?